native/cocos/renderer/core/PassUtils.cpp

```diff
@@ -362,4 +362,34 @@ MaterialProperty toMaterialProperty(gfx::Type type, const ccstd::vector<float> &
     return ret;
 }
 
+bool macroRecordAsBool(const MacroRecord::mapped_type &v) {
+    if (ccstd::holds_alternative<bool>(v)) {
+        return ccstd::get<bool>(v);
+    }
+
+    if (ccstd::holds_alternative<int32_t>(v)) {
+        return ccstd::get<int32_t>(v);
```

Comment:
It returns a bool value?

Reply:
Yes, treat int as a bool, it's moved from ProgramLib.cpp.

Effect data may store `0` or `1` for bool type.

Reply:
I mean should change it explicitly, for example
```c++
return ccstd::get<int32_t>(v) == 0;
```

Reply:
yep, it's better to . I just moved it.

Reply:
DONE